pkg/scheduler/core/generic_scheduler.go

```diff
@@ -361,14 +368,45 @@ func (g *genericScheduler) findNodesThatPassExtenders(pod *v1.Pod, feasibleNodes
 			return nil, err
 		}
 
+		for failedNodeName, failedMsg := range failedAndUnresolvableMap {
+			if _, found := statuses[failedNodeName]; !found {
+				statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, failedMsg)
+			} else {
+				if statuses[failedNodeName].Code() != framework.UnschedulableAndUnresolvable {
+					// UnschedulableAndUnresolvable takes precedence over Unschedulable
+					statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, statuses[failedNodeName].Reasons()...)
+				}
+				statuses[failedNodeName].AppendReason(failedMsg)
+			}
+		}
+
 		for failedNodeName, failedMsg := range failedMap {
+			if _, found := failedAndUnresolvableMap[failedNodeName]; found {
+				// failedAndUnresolvableMap takes precedence over failedMap
+				continue
+			}
 			if _, found := statuses[failedNodeName]; !found {
 				statuses[failedNodeName] = framework.NewStatus(framework.Unschedulable, failedMsg)
 			} else {
 				statuses[failedNodeName].AppendReason(failedMsg)
 			}
 		}
-		feasibleNodes = feasibleList
+
+		prevFailedButResolvableNodes = make(map[string]*v1.Node)
+		for _, node := range feasibleNodes {
+			if status, ok := statuses[node.Name]; ok && status.Code() == framework.Unschedulable {
```

Comment:
How can L396 and L397 happen at the same time? I mean a feasibleNode shouldn't have a Unschedulable or UnschedulableAndUnresolvable status, right?

In the upstream, a failed node won't appear in feasibleNodes, and vice verse - a successful node won't appear in the statusMap:

https://github.com/kubernetes/kubernetes/blob/db9f1e91d977c45fe7df8e1e1bd415a3292995e1/pkg/scheduler/core/generic_scheduler.go#L306-L320

Reply:
OK, IIUC, you're passing the unresolvable nodes to feasible nodes just to give the extender a chance to potentially mark it as UnschedulableAndUnresolvable? if so, it's actually breaking the semantics of "feasible nodes", which is not backward compatible IMO - current extender users wouldn't be surprised that they receive an infeasible node in a feasible nodes list.

Reply:
done